File: CheckPrimeFast.py
# Import the necessary math functions
from math import ceil, sqrt

# A class
class TestPrimenessFast:

    # Constructor
    def __init__(self, suspect):
        self.prime_suspect = suspect

    # returns true if prime_suspect is a prime false otherwise.
    def verifyPrimeFast(self):
        # prime_suspect is a prime number until proven otherwise
        # Loop through searching for factors.
        self.test_range = ceil(sqrt(self.prime_suspect))
        for self.count in range(2, self.test_range):
            if self.prime_suspect % self.count == 0:
                self.a_factor = self.count
                return False

        # If no factor is found:
        return True

File: CheckPrimeFast.cpp
#include "stdafx.h"
#include "CheckPrimeFast.h"


CheckPrimeFast::CheckPrimeFast(unsigned val)
{
	result = "";
	prime_suspect = val;
	square_root = sqrt(prime_suspect); // Get square root
	test_range = (int)ceil(square_root); // Extract an absolute value
}


/**
* Fast copy.
* @param - Takes no parameters
* @return - String (Resolve to check.)
*/
string CheckPrimeFast::verifyPrimeFast() {
	aux.str("");
	aux << prime_suspect;
	result = "Prime State:\n" + aux.str() + " is a prime number.";
	/* Loop through a small range searching for factors. */
	for (int i = 2; i < test_range; i++) {
		if ((prime_suspect % i) == 0) {
			result = "Prime State:\n";
			result += aux.str() + " is not a prime number.\n";
			result += "At least one factor of " + aux.str();
			aux.str("");
			aux << i;
			result += " is " + aux.str();
			break;
		}
	}
	// If no factor is found:
	return result;
}


CheckPrimeFast::~CheckPrimeFast()
{
}
